core/src/main/java/org/apache/calcite/schema/Schemas.java

```diff
@@ -224,15 +224,15 @@ public static DataContext createDataContext(
    * array. */
   public static Enumerable<Object[]> enumerable(final FilterableTable table,
       final DataContext root) {
-    return table.scan(root, ImmutableList.of());
+    return table.scan(root, new ArrayList<>());
```

Comment:
```suggestion
    return table.scan(root, Collections.emptyList());
```

Reply:
No. We need a mutable list. Try running with `-Pguava.version=28.2-jre` and you will see.

Reply:
Why do you think mutable list is required here?
It was working with `ImmutableList.of()`, so it means it should be perfectly fine to pass `Collections.emptyList()`.

Reply:
The javadoc says "the list of filters is mutable".

Reply:
javadoc does not say the implementation can **add** new filters to the list. So `Collections.emptyList()` is fine.